src/Workspaces/Core/Portable/FindSymbols/SymbolFinder_Hierarchy.cs

```diff
@@ -301,37 +301,38 @@ private static bool IsAccessible(SymbolAndProjectId symbolAndProjectId)
         public static async Task<IEnumerable<SymbolCallerInfo>> FindCallersAsync(ISymbol symbol, Solution solution, IImmutableSet<Document> documents, CancellationToken cancellationToken = default)
         {
             symbol = symbol.OriginalDefinition;
-            var foundSymbol = await SymbolFinder.FindSourceDefinitionAsync(symbol, solution, cancellationToken).ConfigureAwait(false);
+            var foundSymbol = await FindSourceDefinitionAsync(symbol, solution, cancellationToken).ConfigureAwait(false);
             symbol = foundSymbol ?? symbol;
 
-            var callReferences = await FindCallReferencesAsync(solution, symbol, documents, cancellationToken).ConfigureAwait(false);
+            var references = await FindCallReferencesAsync(solution, symbol, documents, cancellationToken).ConfigureAwait(false);
 
-            var directReferences = callReferences.Where(
+            var directReference = references.Where(
                 r => SymbolEquivalenceComparer.Instance.Equals(symbol, r.Definition)).FirstOrDefault();
 
-            var indirectReferences = callReferences.WhereAsArray(r => r != directReferences);
+            var indirectReferences = references.WhereAsArray(r => r != directReference);
 
-            List<SymbolCallerInfo> results = null;
+            var results = new List<SymbolCallerInfo>();
 
-            if (directReferences != null)
+            if (directReference != null)
             {
-                foreach (var kvp in await directReferences.Locations.FindReferencingSymbolsAsync(cancellationToken).ConfigureAwait(false))
-                {
-                    results ??= new List<SymbolCallerInfo>();
-                    results.Add(new SymbolCallerInfo(kvp.Key, symbol, kvp.Value, isDirect: true));
-                }
+                await AddReferencingSymbols(directReference, isDirect: true).ConfigureAwait(false);
             }
 
+            foreach (var indirectReference in indirectReferences)
             {
-                var indirectLocations = indirectReferences.SelectMany(r => r.Locations);
-                foreach (var kvp in await indirectLocations.FindReferencingSymbolsAsync(cancellationToken).ConfigureAwait(false))
+                await AddReferencingSymbols(indirectReference, isDirect: false).ConfigureAwait(false);
+            }
+
+            return results;
+
+            async Task AddReferencingSymbols(ReferencedSymbol reference, bool isDirect)
+            {
+                var result = await reference.Locations.FindReferencingSymbolsAsync(cancellationToken).ConfigureAwait(false);
+                foreach (var (callingSymbol, locations) in result)
                 {
-                    results ??= new List<SymbolCallerInfo>();
-                    results.Add(new SymbolCallerInfo(kvp.Key, symbol, kvp.Value, isDirect: false));
+                    results.Add(new SymbolCallerInfo(callingSymbol, reference.Definition, locations, isDirect));
```

Comment:
this line is the primary change.  Previously, we would say that these locations were references to the `symbol` passed into `FindCallersAsync` (modulo getting the `original source definition`).  However, this wasn't really true as these were often references to something like an interface method (or a base/override).  Now we actually pass along the real symbol that was referenced.

Reply:
Can you share a before and after screenshot for some sample code?

Reply:
@mavasani This actually only affects callers of the API.  All features inside Roslyn already only filter to the 'direct' references.  So this is only about what people see when they care about 'indirect references'.